packages/core/test/integration/query-interface/show-all-tables.test.ts

```diff
@@ -0,0 +1,169 @@
+  };
+
+  const queryTableNamesAndNormalizeResults = async (queries: Array<Promise<string[]>>) => {
+    const queryResults = await Promise.all(queries);
+
+    return queryResults.map(normalizeTableNames);
+  };
+
+  it('should not contain views', async () => {
+    async function cleanup(_sequelize: Sequelize) {
+      if (dialectName === 'db2') {
+        // DB2 does not support DROP VIEW IF EXISTS
+        try {
+          await _sequelize.query('DROP VIEW V_Fail');
+        } catch (error: any) {
+          // -204 means V_Fail does not exist
+          // https://www.ibm.com/docs/en/db2-for-zos/11?topic=sec-204
+          if (error.cause.sqlcode !== -204) {
+            throw error;
+          }
+        }
+      } else {
+        await _sequelize.query('DROP VIEW IF EXISTS V_Fail');
+      }
+    }
+
+    await queryInterface.createTable('my_test_table', {
+      name: DataTypes.STRING,
+    });
+    await cleanup(sequelize);
+    const sql = `CREATE VIEW V_Fail AS SELECT 1 Id${
+      ['db2', 'ibmi'].includes(dialectName) ? ' FROM SYSIBM.SYSDUMMY1' : ''
+    }`;
+    await sequelize.query(sql);
+    const tableNames = normalizeTableNames(await queryInterface.showAllTables());
+
+    await cleanup(sequelize);
+    expect(tableNames).to.deep.equal(['my_test_table']);
+  });
+
+  if (!['sqlite', 'postgres', 'db2', 'ibmi'].includes(dialectName)) {
+    // NOTE: sqlite doesn't allow querying between databases and
+    // postgres requires creating a new connection to create a new table.
+    it('should not show tables in other databases', async () => {
+      await queryInterface.createTable('my_test_table1', {
+        name: DataTypes.STRING,
+      });
+      await sequelize.query('CREATE DATABASE my_test_db');
+      await sequelize.query(
+        `CREATE TABLE my_test_db${
+          dialectName === 'mssql' ? '.dbo' : ''
+        }.my_test_table2 (id INT)`,
+      );
+      const tableNames = normalizeTableNames(await queryInterface.showAllTables());
+      await sequelize.query('DROP DATABASE my_test_db');
+      expect(tableNames).to.deep.equal(['my_test_table1']);
+    });
+  }
+
+  if (['mysql', 'mariadb'].includes(dialectName)) {
+    it('should show all tables in all databases', async () => {
+      await queryInterface.createTable('my_test_table1', {
+        name: DataTypes.STRING,
+      });
+      await sequelize.query('CREATE DATABASE my_test_db');
+      await sequelize.query('CREATE TABLE my_test_db.my_test_table2 (id INT)');
+      const tableNames = normalizeTableNames(await sequelize.query(
+        queryInterface.queryGenerator.showTablesQuery(),
+        {
+          raw: true,
+          type: QueryTypes.SHOWTABLES,
+        },
+      ));
+      await sequelize.query('DROP DATABASE my_test_db');
+      expect(tableNames).to.deep.equal(['my_test_table1', 'my_test_table2']);
+    });
+  }
+
+  describe('schema support tests', () => {
+    if (!dialect.supports.schemas || dialectName !== 'postgres') {
```

Comment:
The assertion will fail if we run this for non postgresql dialects that do not support schema in showAllTables.

Reply:
Can you add a TODO that support for other dialects should be added? Or you can add the support for it in this PR directly, I'm fine with both

Reply:
I'll try to add schema in showAllTables support for DB2 & MSSQL other Dialects seem to support schema in the showTablesQuery.

The plan is to update:

DB2
1. Update the [Query generator](https://github.com/sequelize/sequelize/blob/main/packages/core/src/dialects/db2/query-generator.js#L190) to use schema argument
2. Since `getDefaultSchema` is not defined for DB2, use USER as the fallback value.

MSSQL
1. Update the [Query generator](https://github.com/sequelize/sequelize/blob/main/packages/core/src/dialects/mssql/query-generator.js#L237) to use schema argument which will be used as the [TABLE_SCHEMA](https://learn.microsoft.com/en-us/sql/relational-databases/system-information-schema-views/columns-transact-sql?view=sql-server-ver16)